internal/gatewayapi/translator.go

```diff
@@ -546,11 +548,129 @@ func (t *Translator) ProcessHTTPRoutes(httpRoutes []*v1beta1.HTTPRoute, gateways
 						}
 					}
 
+					// TODO: implement header modifier filter
+					for _, filter := range rule.Filters {
+						if irRoute.DirectResponse != nil {
+							// If an invalid filter type has been configured then an error direct response must be returned
+							// so skip processing any other filters for this route
+							break
+						}
+						switch filter.Type {
+						case "RequestRedirect":
+							// Can't have two redirects for the same route
+							if irRoute.Redirect != nil {
+								parentRef.SetCondition(
+									v1beta1.RouteConditionResolvedRefs,
+									metav1.ConditionFalse,
+									v1beta1.RouteReasonUnsupportedValue,
+									"Cannot configure multiple requestRedirect filters for a single HTTPRouteRule",
+								)
+								continue
+							}
+
+							irRedirect := &ir.Redirect{}
+							if redirect := filter.RequestRedirect; redirect != nil {
+								if redirect.Scheme != nil {
+									// Note that gateway API may support additional schemes in the future, but unknown values
+									// must result in an UnsupportedValue status
+									if *redirect.Scheme == "http" || *redirect.Scheme == "https" {
+										irRedirect.Scheme = redirect.Scheme
+									} else {
+										errMsg := fmt.Sprintf("Scheme: %s is unsupported, only 'https' and 'http' are supported", *redirect.Scheme)
+										parentRef.SetCondition(
+											v1beta1.RouteConditionResolvedRefs,
+											metav1.ConditionFalse,
+											v1beta1.RouteReasonUnsupportedValue,
+											errMsg,
+										)
+										continue
+									}
+								}
+
+								if redirect.Hostname != nil {
+									if valid, reason := isValidHostname(string(*redirect.Hostname)); !valid {
+										parentRef.SetCondition(
+											v1beta1.RouteConditionResolvedRefs,
+											metav1.ConditionFalse,
+											v1beta1.RouteReasonUnsupportedValue,
+											reason,
+										)
+										continue
+									} else {
+										redirectHost := string(*redirect.Hostname)
+										irRedirect.Hostname = &redirectHost
+									}
+								}
+
+								if redirect.Path != nil {
+									switch redirect.Path.Type {
+									case "ReplaceFullPath":
```

Comment:
nit, there are constants for the path type options upstream as well (https://github.com/kubernetes-sigs/gateway-api/blob/main/apis/v1beta1/httproute_types.go#L759-L775)